src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -431,28 +432,116 @@ private static bool OverloadResolutionResultIsValid<TMember>(ArrayBuilder<Member
+                        // If we're at a location where the unmanaged data has actually not yet been bound, it cannot be
+                        // valid code anyway, as the only place that can occur is when we're actually doing an address of
+                        // in an attribute.
+                        continue;
+                    }
+
+                    if (unmanagedCallersOnlyData is null)
+                    {
+                        if (!member.CallingConvention.IsCallingConvention(expectedConvention.CallKind)
+                            || !expectedConvention.CallingConventionTypes.IsEmpty)
+                        {
+                            results[i] = makeWrongCallingConvention(result);
+                        }
+
+                        continue;
+                    }
+
+                    Debug.Assert(unmanagedCallersOnlyData.CallingConventionTypes.All(
+                        u => FunctionPointerTypeSymbol.IsCallingConventionModifier((NamedTypeSymbol)u)));
+
+                    // If there is only one type in the unmanagedData, and that type is one of the 4 special names known to the
+                    // compiler, then we treat the CallKind of the method as if it were that calling convention. Otherwise, we
+                    // treat the CallKind as Unmanaged and do comparison from the calling convention types expected in the function
+                    // pointer.
+                    if (unmanagedCallersOnlyData.CallingConventionTypes.Count == 1)
+                    {
+                        switch (unmanagedCallersOnlyData.CallingConventionTypes.Single().Name)
+                        {
+                            case "CallConvCdecl":
+                                _ = checkConventionAgainstActual(Cci.CallingConvention.CDecl, in expectedConvention);
+                                continue;
+                            case "CallConvStdcall":
+                                _ = checkConventionAgainstActual(Cci.CallingConvention.Standard, in expectedConvention);
+                                continue;
+                            case "CallConvThiscall":
+                                _ = checkConventionAgainstActual(Cci.CallingConvention.ThisCall, in expectedConvention);
+                                continue;
+                            case "CallConvFastcall":
+                                _ = checkConventionAgainstActual(Cci.CallingConvention.FastCall, in expectedConvention);
+                                continue;
+                        }
+                    }
+
+                    if (!checkConventionAgainstActual(Cci.CallingConvention.Unmanaged, in expectedConvention))
                     {
-                        results[i] = new MemberResolutionResult<TMember>(
-                            result.Member, result.LeastOverriddenMember,
-                            MemberAnalysisResult.WrongCallingConvention());
+                        return;
```

Comment:
>return [](start = 24, length = 6)

Why return here rather than processing remaining results? Same question for `return;` statements in `switch` below.